src/amuse/datamodel/particle_attributes.py

```diff
@@ -691,6 +691,20 @@ def LagrangianRadii(stars, unit_converter=None, mf=[0.01,0.02,0.05,0.1,0.2,0.5,0
     0.856966667972 length
     """
     import bisect
+    if unit_converter is "auto":
+        # Try to determine the right unit base, using the mass
+        mass_base_unit = stars.mass.unit.base[0][1]
+        if mass_base_unit is units.kg:
+            # Set converter lengths to be something that seems reasonable
+            converter_mass = stars.total_mass() / len(stars)
+            converter_length = (stars.position - stars.center_of_mass()).lengths().mean()
```

Comment:
virial_radius() could also be used..

Reply:
Yes that makes sense. Do you think that's a better choice? It's a little more expensive but probably not noticeably so.